tests/services/logging.service.spec.ts

```diff
@@ -0,0 +1,64 @@
+import { LoggingService } from '../../src/app/core/services/logging.service';
+
+let loggingService: LoggingService;
+const getFilteredLogCount: (currentLog: string, predicate: (line: string) => boolean) => number = (
+  currentLog: string,
+  predicate: (line: string) => boolean
+) =>
+  loggingService
```

Comment:
We should minimise usage of private methods and implementation details in the tests.
This will make it easier to refactor `LoggingService` in future, and the tests become more representative of how the service is used.

Currently, the tests use the internal `getTrimmedLogCache` method directly, and they rely on implementation details
(e.g. how to determine the number of sessions logged, by parsing the whole logs' string)

We could restructure the tests as follows:
- Mock `localStorage` using Jasmine
- As far as possible, only use the public API of LoggingService (i.e. `.startSession()` and `.reset()` to simulate new sessions, `.info()` / `.error()` to add logs)
- Match the stored log against an expected log (instead of comparing the number of logs, as done currently)


@ptvrajsk do you think we can restructure the tests now, or in another PR?

Given that this PR is urgent, I think it is OK to restructure the tests in another PR.

Reply:
@anubh-v

Yes I think I'll make a separate draft PR to restructure the test case so we can take our time to properly implement that.